apps/wallet/src/state/app.ts

```diff
@@ -35,10 +44,44 @@ export const appState = proxy<AppState>({
 	config: defaultConfig,
 	remoteConfig: defaultRemoteConfig,
 	endpoints: defaultEndpoints,
+	navigationDisplay: {
+		navigationHeaderActive: false,
+		drawerPermanent: false,
+		bottomTabActive: false,
+		sidebarAvatarActive: false,
+	},
+	isMobileDisplay: false,
 });
 
 export const appActions = {
 	bootstrap,
 	launchApp,
 	initAfterSignIn,
 };
+
+export const controlNavigationDisplay = () => {
+	const { width } = dimensionState.windowSize;
+	const isResponsiveWidth = width <= 420;
+
+	const isMobileLayout =
+		!runtime.isExtension && (isResponsiveWidth || runtime.isMobile);
+	appState.isMobileDisplay = isMobileLayout;
+
+	if (isMobileLayout) {
+		appState.navigationDisplay = {
+			navigationHeaderActive: true,
+			drawerPermanent: false,
+			bottomTabActive: true,
+			sidebarAvatarActive: false,
```

Comment:
for boolean/flag variable, highly recommend `is..` prefix - e.g `isPermanentDrawer`.